### DISCUSSION TITLE
Enhancement: Various issues (courtesy of epic pro rhythm gamers)

### DISCUSSION THREAD

**Discussion by Epictyphlosion** [FunkinCrew/Funkin#1613] [Ideas]
## What is your suggestion, and why should it be implemented?
Note: I am in no way a veteran at rhythm games. I've simply compiled some complaints and nitpicks made by avid rhythm gamers on Youtube. In other words, I probably have no idea what I'm talking about.

1. The chart editor
From what I and others have observed, the editor is extremely basic compared to most other rhythm games. Because of this, some songs will be difficult or completely impossible to chart, making modding a pain. Features that it is currently missing include:
-BPM higher than 250
-BPMs that are decimal points (e.g. 140.5, 219.75, 179.99)
-Different time signatures
-Different scroll speeds
-Support for more note quantizations

2. Controls and options
I have noticed an options menu is in the works, which is a step in the right direction. Apart from controls, FPS options would be another good addition.

3. Input timing issues
(I have heard by some people that this has been improved, so take this with a grain of salt.) The overall input timing when hitting notes is very poor. Meaning even if you hit every note pixel perfect, there is a chance some won't register as being 100% accurate. This can make FCing a very fast chart extremely difficult. The "Vs. Whitty" mod fixes this, though it involved a major overhaul of the input system, and I believe you (the creator of FNF) stated somewhere that integrating code from mods into the base game is out of the question.

**Comment by PrincessMtH** [comment]
I'm assuming you gathered this list from Koneko's video "[friday night funkin is causing me pain](https://youtu.be/p3qGehBxfos)", as this is the only place I've actually seen that list of issues mentioned, and I'd just like to clarify a few things.

Firstly I would like to confirm that the chart editor sucks ass. It is not very user friendly and features are essentially only added to it when they are needed for the main game, because it wasn't ever aimed at being used by modders to begin with. When the game does eventually feature an actually competent charting tool, it will most definitely be a new one created from scratch. That said, this list of issues is not entirely accurate, so I would like to offer some corrections as well as some updates on what features will be available in the future:

- To my knowledge, there are no issues with having a BPM of 250 or higher. The chart editor does currently limit you to BPMs between 0 and 340, but later versions of the game will allow for BPMs of up to 999.
- Decimal BPMs are currently unsupported, but they will be supported in the next update.
- Different time signatures sadly remain unsupported, as the game's current chart system heavily relies on having 16 steps and 4 beats in a section. Once a proper chart editor is implemented, I expect this to no longer be the case, but I have no idea how long it could be until then.
- Different scroll speeds have always been supported in the game, and most charts have their own scroll speeds set. I can only assume the person who wrote this list meant that the user does not get to choose their own scroll speed as they are specific to the charts, which is true.
- Note quantizations are in a similar position to time signatures, though you can place unquantized notes in the editor by holding the Shift key.
- Another feature mentioned in the original video was support for BPM changes, which has already been implemented in the current public source code.

Additionally, I would like to confirm that the next version will at least increase, but hopefully entirely remove the FPS cap on platforms except HTML5, and that issues with the input system have largely been resolved.
